cmd/snap/cmd_info.go

```diff
@@ -481,27 +468,41 @@ func (x *infoCmd) Execute([]string) error {
 		}
 		noneOK = false
 
-		fmt.Fprintf(w, "name:\t%s\n", both.Name)
-		printSummary(w, both.Summary, termWidth)
-		fmt.Fprintf(w, "publisher:\t%s\n", longPublisher(esc, both.Publisher))
-		if both.Contact != "" {
-			fmt.Fprintf(w, "contact:\t%s\n", strings.TrimPrefix(both.Contact, "mailto:"))
+		if direct != nil {
+			fmt.Fprintf(w, "path:\t%q\n", snapName)
+		}
+		fmt.Fprintf(w, "name:\t%s\n", theSnap.Name)
+		printSummary(w, theSnap.Summary, termWidth)
+		if direct == nil {
+			fmt.Fprintf(w, "publisher:\t%s\n", longPublisher(esc, theSnap.Publisher))
+		} else {
+			// NotesFromRemote might be better called NotesFromNotInstalled but that's nasty
+			fmt.Fprintf(w, "version:\t%s %s\n", theSnap.Version, NotesFromRemote(theSnap, nil))
+			if !osutil.IsDirectory(snapName) {
```

Comment:
I don't like that we have direct checks like this but also maybe* helpers. I think slightly would prefer more maybe helpers

Reply:
maybe another one for later though

Reply:
that was my main pain point reviewing this though, if it cannot be addressed now (after so long) then probably we should close this until we are less busy

Reply:
so this seems to be a dep for the health check work, I still think the new even bigger mix of if and helpers is painful, we do need to do something

Reply:
ok, I'll give it a go in this. I didn't want to make it bigger :-) but, fair enough